src/main/java/csci310group11/Implementation/CollageGenerator.java

```diff
@@ -1,22 +1,109 @@
 package csci310group11.Implementation;
 
-import javq.io.*;
-import java.awt.image.*;
-import javax.imageio.*;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+
+import javax.imageio.ImageIO;
+
+import javafx.scene.layout.BorderImage;
 
 public class CollageGenerator {
-	
-	//download collage to the server
-	public void downloadCollage(Collage collage) {
-		try {
-			BufferedImage image = collage.collage; //getter?
-			String fileName = collage.topic;
-			fileName+= “_” + collage.id  + “.png”;    //if collage has a unique id?
-			File outputFile = new File(topic);
-			ImageIO.write(image, “png”, outputFile);
+	private ArrayList<String> imageUrls; //change to <BufferedImage> if necessary
+	private ArrayList<BufferedImage> images;
+	private ArrayList<BufferedImage> borderedImages;
+	private BufferedImage collage;
+
+	public CollageGenerator(ArrayList<String> urls) {
+		this.imageUrls = new ArrayList<String>();
+		this.images = new ArrayList<BufferedImage>();
+		this.borderedImages = new ArrayList<BufferedImage>();
+		this.collage = new BufferedImage();
+
+		this.imageUrls = urls;
+		for(int i=0; i < imageUrls.size(); i++) {
+			URL url = new URL(imageUrls.get(i));
+			this.images.add(ImageIO.read(url));
 		}
-		catch(Exception e) {
-			e.printStackTrace();
+		this.resizeImages();
+		this.addBorderToImages();
+		this.compileCollage();
+	}
+
+	/**
+	 * Getter for the collage image
+	 * 
+	 * @return the BufferedImage collage
+	 */
+	public BufferedImage getCollage() {
+		return this.collage;
+	}
+
+	/**
+	 * Resizes all the BufferedImages inside of images to be 1/20th of the size of the overall collage.
+	 * 
+	 * Takes the original image and draws it into a new BufferedImage with the proper dimensions
+	 * 
+	 * Notes:
+	 * Should we set collage size 1st or make collage based of images entered
+	 */
+	private void resizeImages() {
+		//1/20th of collage dimensions
+		int resizeWidth = this.collage.getWidth()/20;
+		int resizeHeight = this.collage.getHeight()/20;
+		
+		//Iterate through all images
+		for(int i=0; i < images.size(); i++) {
+			BuffereImage img = images.get(i);
+			//New BufferedImage with 1/20th dimensions of collage
+			BufferedImage resizeImg = new BufferedImage(resizeWidth, resizeHeight, img.getType());
+
+			//Draws the img image into the size of the resizeImg
+			Graphics2D graphics = resizeImg.getGraphics();
```

Comment:
need casting for resizeImg.getGraphics();

Reply:
what casting is required?

Reply:
(Graphics2D)